src/extensions/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,45 @@
+// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Kiota.Abstractions.Extensions
+{
+    /// <summary>
+    /// Extension methods for <see cref="IEnumerable"/>
+    /// </summary>
+    public static class IEnumerableExtensions
+    {
+        public static List<T> AsList<T>(this IEnumerable<T> e) => new List<T>(e);
+
+        public static T[] AsArray<T>(this IEnumerable<T> e)
+        {
+            if (e is ICollection<T> collection)
+            {
```

Comment:
If the source is already an array, we should probably just cast and return that.